### PR TITLE
MOB 1154 disconnect dApps V1 that have no domain

### PR DESCRIPTION
**Pull request by rommex**

### CODE REVIEW

**Comment by rommex** [comment]
@Oleg-Pecheneg made 2 necessary commits, please review

File: unstoppable-ios-app/domains-manager-ios/Modules/QRScanner/ConnectedAppsList/ConnectedAppsListViewPresenter.swift [hunk 1]
```diff
@@ -84,20 +84,21 @@ private extension ConnectedAppsListViewPresenter {
...
-                        }
-                        return nil
+                let items: [ConnectedAppsListViewController.Item] = apps.map({ app in
+                    let domain: DomainItem
+                    if let _domain = domains.first(where: { $0.name == app.domain.name }) {
+                        domain = _domain
+                    } else {
+                        Debugger.printFailure("Forced to display a domain that has been disconnected, \(app.domain.name)", critical: true)
+                        domain = DomainItem(name: "Disconnected: \(app.domain.name)")
                     }
```
Comment (Oleg-Pecheneg) [inline unstoppable-ios-app/domains-manager-ios/Modules/QRScanner/ConnectedAppsList/ConnectedAppsListViewPresenter.swift:94]: We can now take domain directly from connected app?

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift [hunk 2]
```diff
@@ -75,14 +78,21 @@ class WalletConnectServiceV2: WalletConnectServiceV2Protocol {
...
     }
-    
+        
     func disconnectAppsForAbsentDomains(from domains: [DomainItem]) {
         Task {
-            let connectedApps = getConnectedApps()
+            let connectedApps = await getConnectedApps()
             for app in connectedApps {
                 if domains.first(where: { $0.name == app.domain.name }) == nil {
                     try? await disconnect(app: app)
```
Comment (Oleg-Pecheneg) [inline unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift:98]: Looks like a duplicate of function above now

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift [hunk 3]
```diff
@@ -381,7 +404,33 @@ class WalletConnectServiceV2: WalletConnectServiceV2Protocol {
...
 
+extension WalletConnectServiceV2: DataAggregatorServiceListener {
+    func dataAggregatedWith(result: DataAggregationResult) {
+        if case .success(let serviceResult) = result,
+           case .domainsUpdated(let validDomains) = serviceResult {
+            
+            let unifiedApps = getAllUnifiedAppsFromCache()
+            let validConnectedApps = unifiedApps.trimmed(to: validDomains)
+            
+            disconnectApps(from: unifiedApps, notIncluding: validConnectedApps)
```
Comment (Oleg-Pecheneg) [inline unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift:415]: Duplicated logic of `getConnectedApps()` ?

**Review by Oleg-Pecheneg — APPROVED** [review]
With couple of notes

File: unstoppable-ios-app/domains-manager-ios/Modules/QRScanner/ConnectedAppsList/ConnectedAppsListViewPresenter.swift (hunk 1 above)
Comment (rommex) [reply unstoppable-ios-app/domains-manager-ios/Modules/QRScanner/ConnectedAppsList/ConnectedAppsListViewPresenter.swift:94]: Yes, we have a property `domain` as `DomainItem`

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift (hunk 3 above)
Comment (rommex) [reply unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift:415]: They share the logic but I wouldn't mix the method that fetches smth with a method that does smth silently

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift (hunk 2 above)
Comment (rommex) [reply unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift]: True 👍re-used

**Pull request merged**